### PR TITLE
Subscribe users to mailchimp on signup

### LINKED ISSUE #350: Add subscriptions to Mailchimp list
We should subscribe people to our mailing list when they sign up.

This should be done with [gibbon](https://github.com/amro/gibbon)

### PR DESCRIPTION
**Pull request by joshsmith**
Unfinished. Blocked. Needs tests.

Closes #350 when done.
- [x] Add merge fields for first and last name
- [x] Write specs
- [x] Add worker
- [x] Write worker specs
- [x] Add to users controller
- [x] Add specs for user creation on API

### CODE REVIEW

File: spec/workers/subscribe_to_mailing_list_worker_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require "rails_helper"
+
+describe SubscribeToMailingListWorker do
+  it "calls the NotifyPusherOCommentImage scenario" do
```
Comment (christopherstyles) [inline spec/workers/subscribe_to_mailing_list_worker_spec.rb]: I think this should be something like `it "calls subscribe on a CodeCorps::Adapters::MailingList"`?

Comment (joshsmith) [inline spec/workers/subscribe_to_mailing_list_worker_spec.rb]: Yeah this was complete goof.

Comment (joshsmith) [inline spec/workers/subscribe_to_mailing_list_worker_spec.rb]: Fixed.

**Comment by christopherstyles** [comment]
This looks great to me, and is exactly what I had in mind from our [conversation](https://codecorps.slack.com/archives/code-corps-api/p1467840904001289) in Slack. Very nicely done! 

I went ahead and set up a mailchimp list, did a walkthrough of creating a user. Everything functioned as was intended. 👍

**Pull request merged**